### PR TITLE
`"phones <homophone>"` => `"phones list <homophone>"`

### PR DESCRIPTION
**Pull request by pokey**
- Fixes https://github.com/knausj85/knausj_talon/issues/987

### CODE REVIEW

**Comment by pokey** [comment]
Note that this is technically still a Cursorless conflict, as `"list whale"` is a Cursorless target: the list containing a token with a hat over a `w`.  Eg it would match `[world]` if `world` had a hat on the `w`.  So `"phones list whale"` could be interpreted as a Cursorless command trying to cycle homophones for `[world]`.  But no one would ever want to do that, because `"phones"`is supposed to be applied to words.  And in my testing Talon seems to prefer the knausj command here.  So I think this change is ok

**Comment by rntz** [comment]
It looks like you did a replacement rather than just changing the phones <word> command?

File: core/homophones/homophones.talon [hunk 1]
```diff
@@ -1,18 +1,18 @@
-phones <user.homophones_canonical>: user.homophones_show(homophones_canonical)
-phones that: user.homophones_show_auto()
+phones list <user.homophones_canonical>: user.homophones_show(homophones_canonical)
+phones list that: user.homophones_show_auto()
```
[the remark this answers is not in the document]
Comment (pokey) [reply core/homophones/homophones.talon:2]: Also a cursorless conflict, and more consistent imo, as this this command is also listing homophones, no?

**Comment by splondike** [comment]
We should go through the [deprecation process](https://github.com/knausj85/knausj_talon/blob/main/core/deprecations.py) for this change, which means keeping both commands working for six months.

If this is annoying I think we could satisfy both sets of users during the deprecation period by defining a new tag in knausj called `homophones_deactivate_legacy_command`. The original `phones <user.homophones_canonical>` etc. commands would have a `not tag: user.homophones_deactivate_legacy_command` matcher up the top. In the cursorless context you'd set that tag to true to deactivate the deprecated commands.

I also thought that perhaps we could override the capture and action in cursorless to do something else, but that just sounds a bit confusing in this case I think.

File: core/homophones/homophones.talon (hunk 1 above)
[the remark this answers is not in the document]
Comment (rntz) [reply core/homophones/homophones.talon:2]: Wait, what does "phones that" do in cursorless? If it's the same as in knausj then we shouldn't need to change it, should we?

**Comment by rntz** [comment]
I'm still confused how *all* of these are cursorless conflicts. `phones <word>`, sure, that one is inevitable; it's already a self-conflict in knausj. But why does every command starting with "phones" now have to start with "phones list"?

**Comment by rntz** [comment]
Note also that the phones commands should now swap the homophones rather than list them if there are only two.

File: core/homophones/homophones.talon (hunk 1 above)
[the remark this answers is not in the document]
Comment (pokey) [reply core/homophones/homophones.talon:2]: The knausj "phones that" is equivalent to cursorless "phones this". Cursorless "phones that" refers to most recent target. So you could say eg "phones air" to target a token, and then "phones that" to target the same token again (eg if there were 3 homophones and you wanted to go to the next one)

**Comment by pokey** [comment]
> I'm still confused how _all_ of these are cursorless conflicts. `phones <word>`, sure, that one is inevitable; it's already a self-conflict in knausj. But why does every command starting with "phones" now have to start with "phones list"?

Was just trying to be consistent.  I think we could prob keep `phones word` and `phones second word left` as they are if you prefer.  Technically `phones word` is a Cursorless conflict, but the behavior should be basically identical, except when there are 3 homophones, in which case Cursorless cycles and knausj shows a list.

**Comment by pokey** [comment]
> We should go through the [deprecation process](https://github.com/knausj85/knausj_talon/blob/main/core/deprecations.py) for this change, which means keeping both commands working for six months.
> 
> If this is annoying I think we could satisfy both sets of users during the deprecation period by defining a new tag in knausj called `homophones_deactivate_legacy_command`. The original `phones <user.homophones_canonical>` etc. commands would have a `not tag: user.homophones_deactivate_legacy_command` matcher up the top. In the cursorless context you'd set that tag to true to deactivate the deprecated commands.
> 
> I also thought that perhaps we could override the capture and action in cursorless to do something else, but that just sounds a bit confusing in this case I think.

Hmm yeah I guess technically we should.  I don't think a ton of people are using this command in this form though.  I think the tag could work, tho I wonder if it will just make things more confusing if it works differently when you're in vscode

**Comment by splondike** [comment]
Hard to know what's being used. I'm not a heavy Talon user but `phones <word>` is the main one I use :D.

Re the tags thing: The deprecation system endlessly nags people to switch to the new command. I reckon people won't stay using the old phones for too long under that pressure. So it shouldn't be confusing I think?

**Comment by nriley** [comment]
From the community meetup — it seems like there is a lot of low hanging fruit to improve homophone handling in community in general. Also some people get misrecognitions with "phones".

**Comment by pokey** [comment]
Just checking in to see where things stand with this one. Can I just add deprecation for old phones and then ship?

**Comment by nriley** [comment]
@AndreasArvidsson and I talked about this again on Saturday though did not come up with any magical solutions so we didn't end up posting on this. I use "phones <homophone>" quite a bit and really don't want to see this break. One other option we did suggest was only deprecating this when Cursorless was active.

**Comment by AndreasArvidsson** [comment]
We talked this over on the community meetup and we think that too many people are used to the old command grammar and we don't want to break it for them.

**Pull request closed without merging**